src/main/scala/com/softwaremill/react/kafka/KafkaActorSubscriber.scala

```diff
@@ -23,17 +22,12 @@ private[kafka] class KafkaActorSubscriber[T](
       processElement(element.asInstanceOf[T])
     case ActorSubscriberMessage.OnError(ex) =>
       handleError(ex)
-    case ActorSubscriberMessage.OnComplete =>
+    case ActorSubscriberMessage.OnComplete | "Stop" =>
       cleanupResources()
   }
 
   private def processElement(element: T) = {
-    try {
-      producer.send(encoder.toBytes(element), partitionizer(element))
-    }
-    catch {
-      case e: Exception => handleError(e)
-    }
+    producer.send(props.encoder.toBytes(element), props.partitionizer(element))
```

Comment:
Right, using supervisor sounds good. I'm only wondering if it is possible/easy to set it with the current set up. May be worth an example (doesn't need to be in this PR though)